models/notification.go

```diff
@@ -700,6 +752,13 @@ func setNotificationStatusReadIfUnread(e Engine, userID, issueID int64) error {
 	return err
 }
 
+func setRepoNotificationStatusReadIfUnread(e Engine, userID, repoID int64) error {
+	_, err := e.Where(builder.Eq{"user_id": userID}.And(builder.Eq{"status": NotificationStatusUnread}).And(
+		builder.Eq{"source": NotificationSourceRepository}).And(builder.Eq{"repo_id": repoID})).
+		Cols("status").Update(&Notification{Status: NotificationStatusRead})
```

Comment:
Not a fan of this indentation and I'm sure that builder.Eq can actually do multiple Eqs at the same time.

```suggestion
	_, err := e.Where(builder.Eq{
		"user_id": userID,
		"status":  NotificationStatusUnread,
		"source":  NotificationSourceRepository,
		"repo_id": repoID,
	}).Cols("status").Update(&Notification{Status: NotificationStatusRead})
```